sycl/source/exception.cpp

```diff
@@ -10,9 +10,105 @@
 #include <CL/sycl/context.hpp>
 #include <CL/sycl/exception.hpp>
 
+#include <cstring>
+
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 
+namespace { // anonymous
+char reserved_for_errorcode[] =
```

Comment:
```suggestion
const char reserved_for_errorcode[] =
```
Could you please also add a comment explaining why this variable is needed?

Reply:
```suggestion
constexpr char reserved_for_errorcode[] =
```

Reply:
`constexpr` now. Thanks.

@romanovvlad - it's just a very simple way make the MMsg larger when initialized with the what_arg, so we have space for the string-terminator + `[error_code]`